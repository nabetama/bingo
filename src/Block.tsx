import React from "react";

const Block: React.FC<{ label: string; isOpen: boolean }> = (props) => {
  return (
    <div
      className={`${
        props.isOpen ? "bg-indigo-50" : "bg-cyan-600"
      } cursor-pointer block `}
      onClick={() => {
        console.log("debug: ", props.label);
      }}
    >
      {props.label}
    </div>
  );
};

export { Block };
