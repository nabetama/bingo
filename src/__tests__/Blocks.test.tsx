import React from "react";
import { render, screen } from "@testing-library/react";

import { Block } from "../Block";

describe("<Block />", () => {
  test("renders block with argument value", () => {
    render(<Block label="dummy" isOpen={false} toggleIsOpen={jest.fn()} />);
    const val = screen.getByText(/dummy/);
    expect(val).toBeInTheDocument();
  });

  test("snapshot", () => {});
});
