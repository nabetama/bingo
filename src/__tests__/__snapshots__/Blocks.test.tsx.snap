// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blocks /> snapshot 1`] = `
<div
  className="grid grid-cols-5 items-center"
>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    5
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    6
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    7
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    8
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    9
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    10
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    11
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    12
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    13
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    14
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    15
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    16
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    17
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    18
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    19
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    20
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    21
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    22
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    23
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    24
  </div>
  <div
    className="bg-indigo-50 cursor-pointer border-dotted border-gray-400 border-2 text-center  active:bg-emerald-200 w-full h-full max-w-sm sm:max-w-md"
    onClick={[Function]}
  >
    25
  </div>
  <button
    className="col-start-3 col-end-3 mt-10 bg-slate-400 rounded-full py-2 px-4"
    onClick={[Function]}
  >
    リセットする
  </button>
</div>
`;
